packages/react-reconciler/src/ReactInternalTypes.js

```diff
@@ -452,6 +452,7 @@ export type Dispatcher = {
 };
 
 export type AsyncDispatcher = {
+  getActiveCache: () => Map<Function, mixed> | null,
   getCacheForType: <T>(resourceType: () => T) => T,
```

Comment:
There's no need for both of these. If we're saying that the implementation detail is a `Map<Function>` and that is exposed through `getActiveCache()`, there's not that much `getCacheForType` can configure and it's unnecessary indirection. `getCacheForType` can just be a utility function that calls `getActiveCache`. (Or we can get rid of getCacheForType completely and just have the callers call `getActiveCache`.) No need to be on the  dispatcher anymore.

Reply:
To preempt "I'll do it in a follow up". Since we know that these dispatchers are messed with by people and we break them we don't want to change this too many times. Even the async cache ones (cc Janka).

Reply:
Like this [`f140f36` (#30736)](https://github.com/facebook/react/pull/30736/commits/f140f363bfe291ddc6e9bed58033e1e76b3b51ae)?

Whatever resolves the cache conflict needs to know the previous dispatcher so it made sense to me to just do the resolution inside `getActiveCache`. I don't know how else to do it if we want `getActiveCache` to return the actual Map (or null) and resolve conflicts in a utility since that utility needs to know the current and previous dispatcher.